components/skeleton/skeleton-mixin.js

```diff
@@ -168,6 +170,55 @@ export const SkeletonMixin = dedupeMixin(superclass => class extends RtlMixin(su
 	constructor() {
 		super();
 		this.skeleton = false;
+		this._skeletonActive = false;
+		this._readyToDisplay = true;
+		this._waitToDisplay = false;
+
+		this._parentSkeleton = new EventSubscriberController(this, 'skeleton', {
+			onSubscribe: this._onSubscribe.bind(this),
+			onUnsubscribe: this._onUnsubscribe.bind(this)
+		});
+	}
+
+	get skeleton() {
+		return this._skeletonActive;
+	}
+
+	set skeleton(val) {
+		const oldVal = this._skeleton;
+		if (oldVal !== val) {
+			this._skeleton = val;
+			this.requestUpdate('skeleton', oldVal);
+			this.requestUpdate('_skeletonActive', oldVal);
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('skeleton')) {
+			this._updateDisplayState();
```

Comment:
I don't think this call to `_updateDisplayState` is necessary since at this point `_waitToDisplay` won't have been changed (yet). When the below call happens to `updateSubscribers`, the controller _could_ call `changeDisplay` at which point it's going to update the display state.